index.js

```diff
@@ -19,15 +19,10 @@ function Observable(value) {
             return value
         }
 
-        listeners.push(listener)
+        var index = listeners.push(listener) - 1
 
         return function remove() {
-            for (var i = 0, len = listeners.length; i < len; i++) {
-                if (listeners[i] === listener) {
-                    listeners.splice(i, 1)
-                    break
-                }
-            }
+            listeners.splice(index, 1)
```

Comment:
this is the wrong index because anyone can call push() and splice() in arbitrary order

```js
var o = Observ();
var remove1 = o(function () {}); // index 0
var remove2 = o(function () {}); // index 1

remove()1; // remove2 is now index 0
var remove3 = o(function () {}); // index 1

remove2(); // actually removes remove3 (LOL).